src/app/components/editors/shared/modal-confirm/modal-confirm.component.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { BsModalRef } from 'ngx-bootstrap';
+import Spy = jasmine.Spy;
+import { ModalConfirmComponent } from './modal-confirm.component';
+import { ModalConfirmModule } from './modal-confirm.module';
+
+
+describe('ModalConfirmComponent', () => {
+  let component: ModalConfirmComponent;
+  let fixture: ComponentFixture<ModalConfirmComponent>;
+  let hideSpy: Spy;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ModalConfirmModule,
+      ],
+      providers: [
+        BsModalRef,
+      ],
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ModalConfirmComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+
+    hideSpy = spyOn(TestBed.get(BsModalRef), 'hide');
+  });
+
+  it('onConfirm() should correctly hide the modal', () => {
+    const nextSpy = spyOn(component.onClose, 'next');
+
+    component.onConfirm();
```

Comment:
better to use a Page Object and trigger these methods by clicking the html elements